hs_dataverse/utils.py

```diff
@@ -0,0 +1,672 @@
+import json
+import xml.etree.ElementTree as ET
+import re
+import copy
+from datetime import datetime
+from pyDataverse.api import Api
+import os
+import sys
+from django.conf import settings
+import googlemaps
+import time
+
+# global variables
+
+false = False
+true = True
+
+# helper functions
+
+
+# if the given string contains a url, returns it. Otherwise, returns the
+# generic hydroshare url
+def find_url(string, alt_url):
+    """
+    Finds the url in string, otherwise returns alt_url
+
+    :param string: a string potentially containing a url
+    :param alt_url: the alternative url to return if string doesn't have one
+    :return: a url
+    """
+    regex = r"(?i)\b((?:https?://|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/" + \
+        ")(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>" + \
+        "]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:'\".,<>?«»“”‘’]))"
+    url = re.findall(regex, string)
+    if len([x[0] for x in url]) == 0:
+        return alt_url
+    else:
+        return [x[0] for x in url][0]
+
+
+def set_field(val):
+    """
+    returns the field_name of the text of the etree value val, or the empty string if null
+
+    :param val: an etree element containing the field information
+    :return: the string contained in val
+    """
+    if val is None or val.text is None:
+        return 'None'
+    else:
+        return val.text
+
+
+# utility functions
+
+def create_metadata_dict(temp_dir):
+    """
```

Comment:
template.json isn't a template. A template is something you can fill in with code values. It is part of the Django templating interface. You're using it as a simple include and then filling in the values. 

what a real template does is to fill in details in a file, by use of templating operators like "{{ template function }}" that embed values from Django into the file that is being generated. 

Consider a rewrite using Django templating. Templating is based upon calling functions that don't have any arguments. So when you write a template function, you need to write it in such a way that it doesn't need arguments. 
Usually, what I do is to put my accessors into resource.something() which the templating program uses fine. 

It's better to embed tags using templating rather than doing tag lookup. This is much more robust.